### PR TITLE
Improve types for CheckMessage subclasses

### PR DESCRIPTION
**Pull request by adamchainz**
# I have made things!

These classes pass all but the first argument through to `CheckMessage`: https://github.com/django/django/blob/e30d6678421b7573a1995f61521f14348c9b2a17/django/core/checks/messages.py#L59-L81

For type checking purposes it's best to redeclare all the arguments so they are type checked.

## Related issues

n/a

### CODE REVIEW

**Comment by sobolevn** [comment]
Hi! Thanks a lot for your PRs.
I've seen them, but we need to solve https://github.com/python/mypy/issues/13422 first 😞

**Comment by adamchainz** [comment]
Aye... I had a crack at running the tests locally to try fix that but it felt rather unclear to me :( Might need a git bisect on mypy?

**Comment by sobolevn** [comment]
I am pretty sure it is https://github.com/python/mypy/pull/11143

File: django-stubs/core/checks/messages.pyi [hunk 1]
```diff
@@ -19,16 +19,16 @@ class CheckMessage:
     def is_silenced(self) -> bool: ...
 
 class Debug(CheckMessage):
-    def __init__(self, *args: Any, **kwargs: Any) -> None: ...
+    def __init__(self, msg: str, hint: Optional[str] = ..., obj: Any = ..., id: Optional[str] = ...) -> None: ...
```
Comment (sobolevn) [inline django-stubs/core/checks/messages.pyi:22]: `level` is covered by the first call argument 👍 
https://github.com/django/django/blob/e30d6678421b7573a1995f61521f14348c9b2a17/django/core/checks/messages.py#L10

**Review by sobolevn — APPROVED** [review]

**Pull request merged**